components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -94,6 +102,34 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
         }
       });
     } catch (error) {
+      console.error(error);
+      if (error instanceof ResponseError) {
+        switch (error.code) {
+          case ErrorCodes.USER_BLOCKED:
+            this.redirectTo(gitpodHostUrl.with({ pathname: "/blocked" }).toString());
+            return;
+          case ErrorCodes.USER_TERMS_ACCEPTANCE_REQUIRED:
+            this.redirectTo(gitpodHostUrl.withApi({
+              pathname: "/tos",
+              search: `mode=update&returnTo=${encodeURIComponent(window.location.toString())}`,
+            }).toString());
+            return;
+          case ErrorCodes.NOT_AUTHENTICATED:
+            if (!error.data) {
+              this.redirectTo(gitpodHostUrl.withApi({
+                pathname: '/login',
+                search: 'returnTo=' + encodeURIComponent(window.location.toString()),
+              }).toString());
+            }
+            return;
+          case ErrorCodes.USER_DELETED:
+            window.location.href = gitpodHostUrl.asApiLogout().toString();
+            return;
+        }
+      }
```

Comment:
@AlexTugarev mentioned that all these errors should be handled in the top-level entrypoint instead. I'll move them (and re-instate the `/first-steps` redirect as well).